tests/components/switch_as_x/test_init.py

```diff
@@ -282,12 +293,96 @@ async def test_device_registry_config_entry_2(
     assert switch_as_x_config_entry.entry_id in device_entry.config_entries
 
     # Remove the wrapped switch from the device
-    entity_registry.async_update_entity(switch_entity_entry.entity_id, device_id=None)
-    await hass.async_block_till_done()
+    with patch(
+        "homeassistant.components.switch_as_x.async_unload_entry",
+    ) as mock_setup_entry:
+        entity_registry.async_update_entity(
+            switch_entity_entry.entity_id, device_id=None
+        )
+        await hass.async_block_till_done()
+    mock_setup_entry.assert_called_once()
+
     # Check that the switch_as_x config entry is removed from the device
     device_entry = device_registry.async_get(device_entry.id)
     assert switch_as_x_config_entry.entry_id not in device_entry.config_entries
 
+    # Check that the switch_as_x config entry is not removed
+    assert switch_as_x_config_entry.entry_id in hass.config_entries.async_entry_ids()
+
+
+@pytest.mark.parametrize("target_domain", PLATFORMS_TO_TEST)
```

Comment:
I'm wondering what the benefit of testing this for every platform

Reply:
It's to make sure the behavior is same in case we make some change in the future where the platforms don't share code.